ml_scraping/src/RF_model.py

```diff
@@ -0,0 +1,174 @@
+
+# save_word_features = open(glob.glob(os.path.join(DATA_DIR,"pickled_algos", "word_features20%.pickle","wb")
+# pickle.dump(word_features, save_word_features)
+# save_word_features.close()
+'''
+[pos_reviews.append(r) for r in posMovie]
+[neg_reviews.append(r) for r in negMovie]
+
+# tokenize pos & neg reviews
+tokenized_pos = [(word_tokenize(r)) for r in pos_reviews]
+tokenized_neg = [(word_tokenize(r)) for r in neg_reviews]
+pos_pos = nltk.pos_tag(tokenized_pos)
+pos_neg = nltk.pos_tag(tokenized_neg)
+
+
+
+# remove stop words
+stop_words = set(stopwords.words("english"))
+lemmatizer = WordNetLemmatizer()
+
+# lemmatize pos & neg reviews
+lematized_pos = [[lemmatizer.lemmatize(word) for word in sentence if word not in stop_words and word not in punctuation] for sentence in pos_pos]
+lematized_neg = [[lemmatizer.lemmatize(word) for word in sentence if word not in stop_words and word not in punctuation] for sentence in pos_neg]
+#print(lematized_neg[:2])
+'''
+# bag of words all data
+# 20%
+'''
+arr_one = np.array(lematized_pos).flatten()
+arr_two = np.array(lematized_neg).flatten()
+combined = []
+def add(text):
+    if type(text) != str:
+        for l in text:
+            add(l)
+    else:
+        combined.append(text)
+
+add(arr_one)
+add(arr_two)
+'''
+# number of unique words
+all_words = nltk.FreqDist(all_words)
+word_features = list( all_words.keys() )[:len(all_words.keys())//5 ]
+# save_word_features = open(os.path.join(DATA_DIR, "picked_algos", "WordFeatures.pickle"),"wb")
+# pickle.dump(word_features, save_word_features)
+# save_word_features.close()
+def find_features(document):
+    words = word_tokenize(document)
+    lem_words = [lemmatizer.lemmatize(w) for w in words]
+    features = {}
+    for w in word_features:
+        features[w] = (w in lem_words)
+
+    return features
+
+featuresets = [(find_features(rev), category) for (rev, category) in documents]
+ 
+random.shuffle(featuresets)
+print(len(featuresets))
+testing_set = featuresets[len(featuresets)//100:]
+training_set = featuresets[:len(featuresets)//100]
+
+x_train = [ [i[0][w] for w in word_features] for i in training_set]  
+y_train = [i[1] for i in training_set]
+
+x_test = [ [i[0][w] for w in word_features] for i in training_set]  
+y_test = [i[1] for i in testing_set]
+
+
+# train RF
+
+# train_test_spit --> https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html
+clf = RandomForestClassifier(n_estimators=100, n_jobs=-1, verbose=1)
+clf.fit(x_train, y_train)
+# RandomForestClassifier = SklearnClassifier(RandomForestClassifier())
+# RandomForestClassifier.train(training_set)
+# print("Random Forest classifier accuracy percent:", (nltk.classify.accuracy(RandomForestClassifier, testing_set))*100)
+
+
+print(classification_report(y_test, clf.predict(x_test)))
+
+save_classifier = open(os.path.join(DATA_DIR, "picked_algos", "RandomForestClassifier.pickle"),"wb")
+pickle.dump(clf, save_classifier)
+save_classifier.close()
```

Comment:
General note:
using the `with` keyword in python saves you from having to close the file
```python
with open('file_name.pkl', 'wb') as f:
    pickle.dump(obj, f)
```